cmd/mayactl/app/command/volume_info.go

```diff
@@ -0,0 +1,177 @@
+			util.CheckErr(options.Validate(cmd), util.Fatal)
+			util.CheckErr(options.RunVolumeInfo(cmd), util.Fatal)
+		},
+	}
+	cmd.Flags().StringVarP(&options.volName, "volname", "n", options.volName,
+		"unique volume name.")
+	return cmd
+}
+
+// Validate verifies the command whether volName is passed or not.
+func (c *CmdVolumeInfoOptions) Validate(cmd *cobra.Command) error {
+	if c.volName == "" {
+		return errors.New("--volname is missing. Please try running [mayactl volume list] to see list of volumes.")
+	}
+	return nil
+}
+
+// TODO : Add more volume information
+// RunVolumeInfo runs info command and make call to DisplayVolumeInfo
+func (c *CmdVolumeInfoOptions) RunVolumeInfo(cmd *cobra.Command) error {
+	annotation := &Annotations{}
+	// GetVolumeAnnotation is called to get the volume controller's info such as
+	// controller's IP, status, iqn, replica IPs etc.
+	err := annotation.GetVolAnnotations(c.volName)
+	if err != nil {
+		return nil
+	}
+	if annotation == nil {
+		fmt.Println("annotation is nil")
+		return nil
+	}
+	if annotation.ControllerStatus != "Running" {
+		fmt.Println("Volume not reachable, ", annotation.ControllerStatus)
+		return nil
+	}
+	// Initiallize an instance of ReplicaCollection, json response recieved from the
+	// controllerIP:9501/v1/replicas is to be parsed into this structure via GetVolumeStats.
+	// An API needs to be passed as argument.
+	collection := client.ReplicaCollection{}
+	controllerClient := client.ControllerClient{}
+	_, err = controllerClient.GetVolumeStats(annotation.ClusterIP+v1.ControllerPort, v1.InfoAPI, &collection)
+	if err != nil {
+		return err
+	}
+
+	c.DisplayVolumeInfo(annotation, collection)
+	return nil
+}
+
+// DisplayVolumeInfo displays the outputs in standard I/O.
+// Currently It displays volume access modes and target portal details only.
+func (c *CmdVolumeInfoOptions) DisplayVolumeInfo(a *Annotations, collection client.ReplicaCollection) error {
+	var (
+		// address, mode are used here as blackbox for the replica info
+		// address keeps the ip and access mode details respectively.
+		address, mode []string
+		replicaCount  int
+		portalInfo    PortalInfo
+	)
+	const (
+		replicaTemplate = `
+================= Replica Details =================
+IP            AccessMode               Status
+{{range $key, $value := .}}
+{{$value.IP}}     {{$value.AccessMode}}                       {{$value.Status}}
+{{end}}
+===================================================
+`
+		portalTemplate = `
+================= Portal Details ==================
+IQN     :   {{.IQN}}
+Volume  :   {{.VolumeName}}
+Portal  :   {{.Portal}}
+Size    :   {{.Size}}
+Status  :   {{.Status}}
+===================================================
+`
+	)
+	portalInfo = PortalInfo{
+		a.Iqn,
+		c.volName,
+		a.TargetPortal,
+		a.VolSize,
+		a.ControllerStatus,
+	}
+	tmpl, err := template.New("VolumeInfo").Parse(portalTemplate)
+	if err != nil {
+		fmt.Println("Can't Parse the template, found error :", err)
```

Comment:
template is to technical to display to the user. Should we probably say - "Error displaying output" and in the next stage "Error displaying volume details"

Reply:
agree :+1: